nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/ammo_rack.dm

```diff
@@ -0,0 +1,375 @@
+			to_chat(user, "<span class='warning'>You were interrupted!</span>")
+			return TRUE
+		load(I, user)
+		return TRUE
+	return ..()
+
+/obj/machinery/ammo_sorter/AltClick(mob/user)
+	. = ..()
+	setDir(turn(src.dir, -90))
+
+/obj/machinery/ammo_sorter/ex_act(severity, target)
+	for(var/obj/item/X in loaded)
+		X.ex_act(severity, target)
+	. = ..()
+
+/obj/machinery/ammo_sorter/multitool_act(mob/living/user, obj/item/I)
+	var/obj/item/multitool/M = I
+	if(!(M.buffer && istype(M.buffer, /obj/machinery/computer/ammo_sorter)))
+		to_chat(user, "<span class='warning'>There is no control console in [M]'s buffer.")
+		return TRUE
+	var/obj/machinery/computer/ammo_sorter/C = M.buffer
+	if(LAZYFIND(C.linked_sorters, src))
+		to_chat(user, "<span class='warning'>This sorter is already linked to [C]!")
+		return TRUE
+	C.linkSorter(src)
+	to_chat(user, "<span class='notice'>You link [src] to [C].")
+	return TRUE
+
+/obj/machinery/ammo_sorter/Destroy()
+	for(var/obj/machinery/computer/ammo_sorter/AS as() in linked_consoles)
+		AS.linked_sorters -= src
+		AS.ui_update()
+	. = ..()
+
+/obj/machinery/ammo_sorter/examine(mob/user)
+	. = ..()
+	if(panel_open)
+		. += "<span class='notice'>It's maintenance panel is open, you could probably add some oil to lubricate it.</span>" //it didnt tell the players if this was the case before.
+	if(jammed)
+		. += "<span class='notice'>It's jammed shut.</span>"	//if it's jammed, don't show durability. only thing they need to know is that it's jammed.
+	else
+		switch(durability)
+			if(71 to 100)
+				. += "<span class='notice'>It doesn't need any maintenance right now.</span>"
+			if(31 to 70)
+				. += "<span class='notice'>It might need some maintenance done soon.</span>"
+			if(11 to 30)
+				. += "<span class='notice'>It could really do with some maintenance.</span>"
+			if(0 to 10)
+				. += "<span class='notice'>It's completely wrecked.</span>"
+	. += "<br/><span class='notice'>It's currently holding [length(loaded)]/[max_capacity] items:</span>"
+	if(length(loaded))
+		var/listofitems = list()
+		for(var/obj/item/C in loaded)
+			var/path = C.type
+			if (listofitems[path])
+				listofitems[path]["amount"]++
+			else
+				listofitems[path] = list("name" = C.name, "amount" = 1)
+		for(var/i in listofitems)
+			. += "<span class='notice'>[listofitems[i]["name"]] x[listofitems[i]["amount"]]</span>"
+
+/obj/machinery/ammo_sorter/RefreshParts()
+	max_capacity = 0
+	for(var/obj/item/stock_parts/matter_bin/MB in component_parts)
+		max_capacity += MB.rating+3
+	if(max_capacity < length(loaded))
+		pop()
+
+/obj/machinery/ammo_sorter/MouseDrop_T(atom/movable/A, mob/user)
+	. = ..()
+	if(!isliving(user))
+		return FALSE
+	//You can store any kind of ammo here for now.
+	if(istype(A, /obj/item/ship_weapon/ammunition) || istype(A, /obj/item/powder_bag))
+		to_chat(user, "<span class='notice'>You start to load [src] with [A].</span>")
+		if(do_after(user, 2 SECONDS , target = src))
```

Comment:
It could be fun to make this speed dependant on a stock part rating (by adding a manipulator?) so it can be upgraded. Not required for this PR though